content/blog/workflowsets-0-0-1/index.Rmd

```diff
@@ -0,0 +1,316 @@
+
+## A Dimension Reduction Example
+
+When a data set is very wide (i.e., many columns), a dimension reduction procedure may be able to represent the data as a smaller subset of new, artificial variables. The most common method for dimensional reduction is principal component analysis (PCA) but there are many others. [Kuhn and Johnson (2018)](https://bookdown.org/max/FES/numeric-many-to-many.html#linear-projection-methods) give an overview of several other techniques that will be used below. 
+
+```{r quiet, include = FALSE}
+library(conflicted)
+
+conflict_prefer("filter",     winner = "dplyr",    quiet = TRUE)
+conflict_prefer("select",     winner = "dplyr",    quiet = TRUE)
+conflict_prefer("slice",      winner = "dplyr",    quiet = TRUE)
+conflict_prefer("rename",     winner = "dplyr",    quiet = TRUE)
+conflict_prefer("tune",       winner = "tune",     quiet = TRUE)
+conflict_prefer("neighbors",  winner = "dials",    quiet = TRUE)
+
+library(MASS)
+library(stats)
+library(tidymodels)
+library(workflowsets)
+library(tidyposterior)
+library(doMC)
+registerDoMC(cores = 20)
+
+theme_set(theme_bw())
+
+data(meats)
+corr_mat <- cor(meats[, 1:100])
+mean_corr <- mean(corr_mat[upper.tri(corr_mat)])
+mean_corr <- round(mean_corr * 100, 1)
+```
+
+There are at least two practical uses of dimension reduction: 
+
+* When assessing the quality of the data, it may be difficult to view it in high dimensions. Reducing the number of columns to a more manageable set helps the user understand and identify problematic data points. 
+
+* Some models are less effective with a large number of predictors (relative to the number of data points). This is especially true when the predictors have a high degree of correlation with one another (e.g., multicolinearity). 
+
+As an example, the [Tecator data](https://modeldata.tidymodels.org/reference/meats.html) set uses spectrography to predict the percentage of fat, protein, and water in meat samples. They measured 100 channels of the spectum for 215 samples. The _average_ correlation between predictors is `r mean_corr`%. These data might greatly benefit from dimension reduction. We'll evaluate a few different approaches to find a smaller representation of the predictors when the _percentage of water_ is being predicted. 
+
+We'll load the packages and data, then define a resampling scheme that is used to evaluate the models.
+
+```{r start}
+library(tidymodels)
+library(workflowsets)
+library(tidyposterior)
+
+data(meats, package= "modeldata")
+
+# Keep only the water outcome
+meats <- select(meats, -fat, -protein)
+
+set.seed(1)
+meat_split <- initial_split(meats)
+meat_train <- training(meat_split)
+meat_test <- testing(meat_split)
+
+set.seed(2)
+meat_folds <- vfold_cv(meat_train, repeats = 3)
+```
+
+Workflow sets take a set of **preprocessors** and **models** and combinatorially combine them into a series of workflow objects. The package has some nice convenience functions to tune and/or evaluate each workflow. 
```

Comment:
```suggestion
The new workflowsets package takes a set of **preprocessors** and **models** and combinatorially combines them into a set of multiple workflow objects. The package has convenience functions to tune and/or evaluate each workflow. 
```

I think we should avoid talking about these sets as a "series" because they don't typically have ordering